# Deep mutational scanning of avian influenza PB2 to comprehensively map adaptation to the human host

Authors: Shirleen Soh, Louise Moncla, Rachel Eguia, Trevor Bedford, Jesse Bloom

### This code is associated with the paper from Soh et al., "Comprehensive mapping of adaptation of the avian influenza polymerase protein PB2 to humans". eLife, 2019. http://dx.doi.org/10.7554/eLife.45079

## Overview

We performed deep mutational scanning of PB2 from avian influenza strain A/Green-winged Teal/Ohio/175/1986 (S009) to comprehensively map the effect of all amino-acid mutations to an avian influenza PB2 on viral replication in different host species.

The computational analyses supporting the manuscript are documented here.


## Organization

### Analysis notebooks

Analyses are organized by figure. Notebooks for Fig1 through Fig7 need to be run in order.

* [`Fig1_ProcessDMSdata.ipynb`](Fig1_ProcessDMSdata.ipynb)

* [`Fig2_InferPreferences.ipynb`](Fig2_InferPreferences.ipynb)

* [`Fig3_IdentifyAdaptiveMutations.ipynb`](Fig3_IdentifyAdaptiveMutations.ipynb)

* [`Fig4_Validation.ipynb`](Fig4_Validation.ipynb)

* [`Fig5_MapDiffSelToStructure.ipynb`](Fig5_MapDiffSelToStructure.ipynb)

* [`Fig6_H7N9.ipynb`](Fig6_H7N9.ipynb)

* [`Fig7_Accessibility.ipynb`](Fig7_Accessibility.ipynb)

* [`FigS1_compareS009.ipynb`](FigS1_compareS009.ipynb)

### Running the notebooks

The notebooks are all run in order using the bash script [run_notebooks.bash](run_notebooks.bash).
To run them on the Hutch cluster, use:

    sbatch -p largenode -c 14 --mem=300000 --time=48:00:00 run_notebooks.bash

### Subdirectories

* [./scripts/](./scripts/) and [./pymodules/](./pymodules/) contains custom scripts and functions used by the analyses.

* [./data/](./data/) contains input data for analysis of deep mutational scanning experiment.

* [./fastq/](./fastq/) contains the deep mutational scanning and validation sequencing data.

* [./results/](./results/) contains results generated by analysis of deep mutational scanning experiment. Most results are not tracked on GitHub, except for key output files described as follows: 

    * The [./results/codoncounts/](results/codoncounts) subdirectory has the counts of each codon from the deep sequencing data in each condition.
 
    * The [./results/prefs_A549/](results/prefs_A549/) and [./results/prefs_CCL141/](results/prefs_CCL141/) subdirectories have the preferences of each amino acid as measured in each cell line.
 
    * The [./results/diffsel/](results/diffsel/) subdirectory has the differential selection values of each amino acid. 

* [./analyzelibrary/](./analyzelibrary/) contains analysis of mutant libraries based on the results from Sanger sequencing some selected clones. Analysis by script [`analyze_library.py`](https://github.com/jbloomlab/SangerMutantLibraryAnalysis).

* [./validation/](./validation/) contains input data and results for validation experiments, as analyzed by [`Fig4_Validation.ipynb`](Fig4_Validation.ipynb).

* [./pymol/](./pymol/) contains pymol scripts used to generate annotated structures of PB2, as generated by [`Fig5_MapDiffSelToStructure.ipynb`](Fig5_MapDiffSelToStructure.ipynb). Also contains PDB files for S009-PB2 threaded through respective structures.

* [./compareS009/](./compareS009/) contains input data and results for phylogenetic analysis of our chosen avian influenza strain in relation to other influenza strains, as generated by [`FigS1_compareS009.ipynb`](FigS1_compareS009.ipynb).

* [./paper/](./paper/) contains figure PDFs used in manuscript figures, as well as supplementary tables submitted with manuscript.
